### DISCUSSION TITLE
How to install runtime?

### DISCUSSION THREAD

**Discussion by Cavebrute** [airsdk/Adobe-Runtime-Support#2726] [General]
Seriously, how do I install this? 
After running AdobeAIR.exe, the only thing visible to my eyes that happens, is it creates another process, then just nothing happens.
Tried all compatibility modes already.

windows 11
Microsoft Windows [Version 10.0.22621.525]

**Comment by ajwfrost** [comment]
Hi

We've just checked this on Windows 11 build 22000.1936 (21H2) and it worked, albeit without displaying the EULA properly. So there must be a configuration difference or a dependency that's missing on your machine. I suspect it's around the web view capabilities (which is what's used for the EULA display)...

Looking closer at this, on our Windows 11 machine, the Add/Remove programs shows that we have the "Microsoft Edge WebView2 Runtime" installed, and yet when we look at the registry for the keys that Microsoft say will be present and provide information about this, they are missing. We use those keys to know where to load in the WebView2 library from (as we can't link statically, otherwise the binary would completely fail to load on older Windows versions).

It does look like the webview is installed into a fairly standard location, so probably we can add a workaround here that means if the registry entries are missing, we go looking for the library where we hope it may be located.. So we would need to do an update to the runtime for this.

As a workaround for now, can you try using a command-line - it would need to have administrator privileges, so if you hit the Windows button and type "command", you should then see a "Run as administrator" option appear on the right half of the menu.
```
AdobeAIR.exe -silent -eulaAccepted
```
This would bypass the UI and should then do the installation quietly. You can check from Add/Remove Programs to see if it's worked (and it should say "done" on the command line too..)

thanks

**Reply by ajwfrost** [reply]
Quick update -> it wasn't actually the "WebView2" that was the problem it was the original "WebView" I believe. That may not be present on your Windows 11 machine.. it's meant to be available until 2029 but it may now be an optional component (mshtml.dll, normally in c:\windows\system32...)

So actually, we would need to move this over to the new WebView2 capability. The registry key was there in the end but it's under the 32-bit registry hence we missed that when first looking via normal/64-bit regedit...

We will get this built and uploaded soon..

thanks

**Reply by Cavebrute** [reply]
I didn't remember having any issues like this on older versions of Windows, so I was kind of lost.
But thank you for your patience and help, the silent installation did the job!
